gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -409,3 +409,109 @@ gis_unattended_config_match_computer (GisUnattendedConfig *self,
 
   return GIS_UNATTENDED_COMPUTER_DOES_NOT_MATCH;
 }
+
+static GFile *
+invent_backup_name (GFile *source)
+{
+  g_autoptr(GFile) dir = g_file_get_parent (source);
+  g_autofree gchar *basename = g_file_get_basename (source);
+  g_autofree gchar *uuid = g_uuid_string_random ();
+  gchar *dot = strrchr (basename, '.');
+  g_autofree gchar *target_basename = NULL;
+
+  if (dot == NULL)
+    {
+      target_basename = g_strjoin (".", basename, uuid, NULL);
+    }
+  else
+    {
+      *dot = '\0';
+      target_basename = g_strjoin (".", basename, uuid, dot + 1, NULL);
+    }
+
+  return g_file_get_child (dir, target_basename);
+}
+
+static gboolean
+rename_existing_file_if_exists (GFile   *source,
+                                gchar  **backup_path,
+                                GError **error)
+{
+  g_autoptr(GFile) target = invent_backup_name (source);
+  g_autoptr(GError) local_error = NULL;
+
+  if (g_file_move (source, target,
+                   /* No G_FILE_COPY_OVERWRITE, so target will not be
+                    * overwritten if it already exists. Since its name includes
+                    * a fresh UUID, this is extremely unlikely.
```

Comment:
So if there *is* a `target` collision, this will error. Should it not retry instead?

Reply:
It should – at which point one may as well use `g_mkstemp()` (which has a retry loop), at which point maybe one may as well use `glnx_renameat2_exchange()`…

Reply:
Well, yes, that sounds good.